pkg/storage/odf_test.go

```diff
@@ -0,0 +1,401 @@
+			assert.Equal(t, testStorageSystemBuilder.Definition.Namespace, testStorageSystemBuilder.Object.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		}
+	}
+}
+
+func TestStorageSystemDelete(t *testing.T) {
+	testCases := []struct {
+		testStorageSystem *StorageSystemBuilder
+		expectedError     error
+	}{
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     nil,
+		},
+		{
+			testStorageSystem: buildInValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     fmt.Errorf("storageSystem 'name' cannot be empty"),
+		},
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:     nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testStorageSystem.Delete()
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testStorageSystem.Object)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestStorageSystemUpdate(t *testing.T) {
+	testCases := []struct {
+		testKind          odfoperatorv1alpha1.StorageKind
+		testName          string
+		testNamespace     string
+		testStorageSystem *StorageSystemBuilder
+		expectedError     error
+	}{
+		{
+			testKind:          odfoperatorv1alpha1.StorageKind("storagecluster.ocs.openshift.io/v1"),
+			testName:          "ocs-storagecluster",
+			testNamespace:     defaultStorageSystemNamespace,
+			testStorageSystem: buildValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     nil,
+		},
+		{
+			testKind:          odfoperatorv1alpha1.StorageKind("storagecluster.ocs.openshift.io/v1"),
+			testName:          "ocs-storagecluster",
+			testNamespace:     defaultStorageSystemNamespace,
+			testStorageSystem: buildInValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     fmt.Errorf("storageSystem 'name' cannot be empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Equal(t, odfoperatorv1alpha1.StorageKind(""), testCase.testStorageSystem.Definition.Spec.Kind)
+		assert.Equal(t, "", testCase.testStorageSystem.Definition.Spec.Name)
+		assert.Equal(t, "", testCase.testStorageSystem.Definition.Spec.Namespace)
+		assert.Nil(t, nil, testCase.testStorageSystem.Object)
+
+		testCase.testStorageSystem.WithStorageClusterSpec(testCase.testKind, testCase.testName, testCase.testNamespace)
+		_, err := testCase.testStorageSystem.Update()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testCase.testKind, testCase.testStorageSystem.Definition.Spec.Kind)
+			assert.Equal(t, testCase.testName, testCase.testStorageSystem.Definition.Spec.Name)
+			assert.Equal(t, testCase.testNamespace, testCase.testStorageSystem.Definition.Spec.Namespace)
+		} else {
+			assert.Equal(t, testCase.expectedError, err)
```

Comment:
Same

Reply:
same

Reply:
Again, I think it should be the same as the other cases

Reply:
this is for an update:
`--- PASS: TestStorageSystemDelete (0.01s)
=== RUN   TestStorageSystemUpdate
    odf_test.go:311: 
                Error Trace:    /home/elgerman/eco-gotest/eco-goinfra-odf/pkg/storage/odf_test.go:311
                Error:          Not equal: 
                                expected: <nil>(<nil>)
                                actual  : *errors.StatusError(&errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"Operation cannot be fulfilled on storagesystems.odf.openshift.io \"ocs-storagecluster-storagesystem\": object was modified", Reason:"Conflict", Details:(*v1.StatusDetails)(0xc00094bb00), Code:409}})
                Test:           TestStorageSystemUpdate
--- FAIL: TestStorageSystemUpdate (0.01s)
`

Reply:
Yeah it seems to be because your Update method doesn't have the force argument. Similar to other resources that use the runtime-client. E.g. https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/cgu/cgu.go#L250

Reply:
This means that update probably won't work in a real environment also

Reply:
but it is working :) this is not the first time I see such error.
but let me see what I can do...

Reply:
@trewest if what you are saying is right, so:
- there is no other option then to add force as hardcoded, not force doesn't work at all; please confirm
- you need to fix TestClusterImageSetUpdate test, since there is no force=true variation included

Reply:
@trewest another option is just to delete this method